packages/core/upload/admin/src/components/AssetDialog/BrowseStep/index.js

```diff
@@ -88,48 +99,58 @@ export const BrowseStep = ({
-                      background="neutral0"
-                      hasRadius
-                      borderColor="neutral200"
-                      height={`${32 / 16}rem`}
-                    >
-                      <BaseCheckbox
-                        aria-label={formatMessage({
-                          id: getTrad('bulk.select.label'),
-                          defaultMessage: 'Select all assets',
-                        })}
-                        indeterminate={!areAllAssetSelected && hasSomeAssetSelected}
-                        value={areAllAssetSelected}
-                        onChange={onSelectAllAsset}
-                      />
-                    </Flex>
-                  )}
-                  <SortPicker onChangeSort={onChangeSort} />
-                  <Filters
-                    appliedFilters={queryObject?.filters?.$and}
-                    onChangeFilters={onChangeFilters}
-                  />
-                </StartBlockActions>
-              )}
+          <Flex justifyContent="space-between" alignItems="flex-start">
+            {(assetCount > 0 || folderCount > 0 || isFiltering) && (
+              <StartBlockActions wrap="wrap">
+                {multiple && (
+                  <Flex
+                    paddingLeft={2}
+                    paddingRight={2}
+                    background="neutral0"
+                    hasRadius
+                    borderColor="neutral200"
+                    height={`${32 / 16}rem`}
+                  >
+                    <BaseCheckbox
+                      aria-label={formatMessage({
+                        id: getTrad('bulk.select.label'),
+                        defaultMessage: 'Select all assets',
+                      })}
+                      indeterminate={!areAllAssetSelected && hasSomeAssetSelected}
+                      value={areAllAssetSelected}
+                      onChange={onSelectAllAsset}
+                    />
+                  </Flex>
+                )}
+                <SortPicker onChangeSort={onChangeSort} />
+                <Filters
+                  appliedFilters={queryObject?.filters?.$and}
+                  onChangeFilters={onChangeFilters}
+                />
+              </StartBlockActions>
+            )}
+
+            {(assetCount > 0 || folderCount > 0 || isSearching) && (
+              <EndBlockActions pullRight>
+                <SearchAsset onChangeSearch={onChangeSearch} queryValue={queryObject._q || ''} />
+              </EndBlockActions>
+            )}
+          </Flex>
+        </Box>
+      )}
 
-              {(assetCount > 0 || folderCount > 0 || isSearching) && (
-                <EndBlockActions pullRight>
-                  <SearchAsset onChangeSearch={onChangeSearch} queryValue={queryObject._q || ''} />
-                </EndBlockActions>
-              )}
-            </Flex>
-          </Box>
+      {canRead && !isCurrentFolderLoading && (
+        <Box paddingTop={3}>
```

Comment:
Should we make the parent `Box` a `Stack space={3}` instead?

Reply:
I did several tests for that, I removed the initial Stack because it didn't fit Figma's designs 
the current code should reflect the design 👀

<img width="724" alt="image" src="https://user-images.githubusercontent.com/71838159/182153783-865240ff-95f0-4ca6-becb-1a1964868af0.png">

Reply:
Ok then :)